### PR TITLE
Small refactor

### PR DESCRIPTION
**Pull request by JoD**
Refactors aimed at improving the quality of the code, so that future work is easier.

### CODE REVIEW

File: roundingsat.cc [hunk 1]
```diff
@@ -644,6 +650,27 @@ void reduce_by_toplevel(vector<int>& lits, vector<int>& coefs, int& w){
 	coefs.erase(coefs.begin()+j,coefs.end());
 }
 
+// UTIL ----------------------------------------------------------------
+inline void saturate(vector<int>& coefs, int& w){
+	for (auto it = coefs.begin(); it != coefs.end(); ++it) if (it [0] > w) it [0] = w;
```
Comment (elffersj) [inline roundingsat.cc]: Please revert to the original. I should not have updated it.
```
for (int i = 0; i < (int) coefs.size(); i++) coefs[i] = min(coefs[i], w);
```

File: roundingsat.cc [hunk 2]
```diff
@@ -644,6 +650,27 @@ void reduce_by_toplevel(vector<int>& lits, vector<int>& coefs, int& w){
...
+	if(w<=0) return true; // trivially satisfied constraint
+	saturate(coefs,w); // as reduce_by_toplevel could have reduced w
+	return false;
+}
+
+CRef learnConstraint(vector<int>& lits,vector<int>& coefs, int w){
+	CRef cr = ca.alloc(lits,coefs,w, true);
+	Clause & C = ca[cr];
+	C.lbd = computeLBD(cr);
+	attachClause(cr);
```
Comment (elffersj) [inline roundingsat.cc]: The disadvantage of this is that sometimes, I don't want to call attachClause for all learnt constraints. But this is only in experiments.

Comment (JoD) [reply roundingsat.cc]: Done

File: roundingsat.cc (hunk 1 above)
Comment (JoD) [reply roundingsat.cc]: Done.

**Comment by JoD** [comment]
I've added a couple more commits.

**Comment by elffersj** [comment]
Great! Will you also refactor `learnConstraint` or do you prefer to merge the current pull req?

**Comment by JoD** [comment]
I'd prefer to merge the current pull request, but it's not something I feel strongly about. If you think it's better without a separate `learnConstraint` method, I'll change it again.

**Pull request merged**